modules/github/validate-repository-settings/main.tf

```diff
@@ -0,0 +1,143 @@
+/**
+  * This module sets up a GitHub repository in a standardized way, to ensure
+  * consistency among things like branch and PR handling, labels and so on.
+  */
+
+terraform {
+  required_version = "0.15.3"
+
+  required_providers {
+    github = {
+      source  = "integrations/github"
+      version = "~> 4.0"
+    }
+  }
+}
+
+provider "github" {
+  owner = "XenitAB"
+  token = var.github_token
+}
+
+resource "github_repository" "this" {
+  name        = var.repository_name
+  description = var.repository_description
+
+  allow_auto_merge       = false
+  allow_merge_commit     = true
+  allow_rebase_merge     = true
+  allow_squash_merge     = true
+  delete_branch_on_merge = true
+  has_downloads          = false
+  has_issues             = true
+  has_wiki               = false
+  visibility             = var.repository_visibility
+}
+
+resource "github_branch_protection" "this" {
+  repository_id  = github_repository.this
+  pattern        = "main"
+  enforce_admins = true
+  required_pull_request_reviews {
+    required_approving_review_count = 1
+  }
+  required_status_checks {
+    contexts = var.required_status_checks
+  }
```

Comment:
Do we want `required_linear_history` as well?
From docs:
required_linear_history - (Optional) Boolean, setting this to true enforces a linear commit Git history, which prevents anyone from pushing merge commits to a branch

Reply:
I think that means you have to use "rebase PRs" or `git merge --ff-only` which I don't think we normally want.

Reply:
Ah yes, reading GitHub's documentation more carefully, I think you are right. Then we don't want this.

As an aside, I have a long-running grudge against GitHub for not supporting the one true workflow that I want, which is "branches must be rebased on tip of `main` before merge, and merge should be performed using `git merge --no-ff`".